src/containers/BookList.js

```diff
@@ -1,39 +1,54 @@
 import React from 'react';
 import Book from '../components/Book';
 import { connect } from 'react-redux';
-import {removeBook} from '../actions/index';
+import {removeBook, changeFilter} from '../actions/index';
+import CategoryFilter from '../components/CategoryFilter'
 import '../components/App.css';
 
 class BookList extends React.Component {
 	
   render (){
-    return (<table>
-    	       <tbody>
-    	       <tr>
-                  <th>ID</th>
-                  <th>TITLE</th>
-                  <th>CATEGORY</th>
-                </tr>
-    	       {this.props.mybooks.map((elm) => {
-    	       	 return (<Book key = {elm.id} value = {elm} 
-    	       	 	      handleRemoveBook = {this.props.delItem} />)
-    	       })}
-               </tbody>
-    	   </table>);
+    let arrBook  = this.props.mybooks;
+    let filterBook = this.props.myFilter;
+    if (filterBook !== 'All'){
+    	arrBook = arrBook.filter(book => book.category === filterBook)
```

Comment:
[OPTIONAL]
- we can avoid passing all the books into the component by having a method that takes the books and filter from the store and will give us just what is needed. that method is what will be passed in mapStateToProps instead.